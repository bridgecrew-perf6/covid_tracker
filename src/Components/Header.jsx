import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHome, faMicrophone, faCog } from '@fortawesome/fontawesome-free-solid';
import './Sass/Header.scss';

function Header() {
  return (
    <nav>
      <Link to="/"><FontAwesomeIcon icon={faHome} /></Link>
      <h3>Coronavirus Tracker</h3>
      <FontAwesomeIcon icon={faMicrophone} />
      <FontAwesomeIcon icon={faCog} />
    </nav>
  );
}

export default Header;
