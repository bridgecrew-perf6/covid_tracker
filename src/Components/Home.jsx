import React from 'react';
import Continents from './Continents';
// import Countries from './Countries';
import './Sass/Styles.scss';

function Home() {
  return (
    <div className="container">
      <Continents />
      {/* <Countries /> */}
    </div>
  );
}

export default Home;
