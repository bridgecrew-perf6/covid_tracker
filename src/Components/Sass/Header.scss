nav {
  display: flex;
  align-items: center;
  padding: 0 4vw 0 4vw;
  gap: 35vw;
  height: 15vh;
  background-color: #db6d85;
  color: #ebebee;
  font-size: larger;
  border-bottom: 2px solid;

  .nav-right {
    display: flex;
    gap: 2vw;
  }
}

@media only screen and (max-width: 600px) {
  nav {
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 15vw;
    width: 100vw;
    height: 15vh;
  }
}
