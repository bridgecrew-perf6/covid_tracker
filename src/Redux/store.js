import { composeWithDevTools } from 'redux-devtools-extension';
import { combineReducers, legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import countriesReducer from './Countries/Reducers/countriesReducer';

const rootReducer = combineReducers({
  countries: countriesReducer,
});

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
