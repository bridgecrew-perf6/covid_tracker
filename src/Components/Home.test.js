import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/store';
import Header from './Header';
import Home from './Home';

describe('Render components', () => {
  test('render Header', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Header /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('render Home', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Home /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
