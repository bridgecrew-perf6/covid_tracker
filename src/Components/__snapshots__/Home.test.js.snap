// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components render Header 1`] = `
<nav>
  <a
    href="/"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-home "
      data-icon="home"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <h3>
    Coronavirus Tracker
  </h3>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-microphone "
    data-icon="microphone"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 352 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-cog "
    data-icon="cog"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M444.788 291.1l42.616 24.599c4.867 2.809 7.126 8.618 5.459 13.985-11.07 35.642-29.97 67.842-54.689 94.586a12.016 12.016 0 0 1-14.832 2.254l-42.584-24.595a191.577 191.577 0 0 1-60.759 35.13v49.182a12.01 12.01 0 0 1-9.377 11.718c-34.956 7.85-72.499 8.256-109.219.007-5.49-1.233-9.403-6.096-9.403-11.723v-49.184a191.555 191.555 0 0 1-60.759-35.13l-42.584 24.595a12.016 12.016 0 0 1-14.832-2.254c-24.718-26.744-43.619-58.944-54.689-94.586-1.667-5.366.592-11.175 5.459-13.985L67.212 291.1a193.48 193.48 0 0 1 0-70.199l-42.616-24.599c-4.867-2.809-7.126-8.618-5.459-13.985 11.07-35.642 29.97-67.842 54.689-94.586a12.016 12.016 0 0 1 14.832-2.254l42.584 24.595a191.577 191.577 0 0 1 60.759-35.13V25.759a12.01 12.01 0 0 1 9.377-11.718c34.956-7.85 72.499-8.256 109.219-.007 5.49 1.233 9.403 6.096 9.403 11.723v49.184a191.555 191.555 0 0 1 60.759 35.13l42.584-24.595a12.016 12.016 0 0 1 14.832 2.254c24.718 26.744 43.619 58.944 54.689 94.586 1.667 5.366-.592 11.175-5.459 13.985L444.788 220.9a193.485 193.485 0 0 1 0 70.2zM336 256c0-44.112-35.888-80-80-80s-80 35.888-80 80 35.888 80 80 80 80-35.888 80-80z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</nav>
`;

exports[`Render components render Home 1`] = `
<div
  className="container"
>
  <form
    className="search-bar"
    onChange={[Function]}
  >
    <div
      className="search-bar__wrapper"
    >
      <input
        className="input"
        placeholder="Country Name"
      />
    </div>
  </form>
  <div
    className="container"
  />
</div>
`;
